third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -2020,33 +2057,29 @@ impl Frame {
                         interpreter.operand_stack.push(field_ref)?;
                     },
                     Bytecode::Pack(sd_idx) => {
-                        let mut get_field_count_charge_gas_and_check_depth =
-                            || -> PartialVMResult<u16> {
-                                let field_count = self.field_count(*sd_idx);
-                                let struct_type = self.get_struct_ty(*sd_idx);
-                                interpreter.ty_depth_checker.check_depth_of_type(
-                                    gas_meter,
-                                    traversal_context,
-                                    &struct_type,
-                                )?;
-                                Ok(field_count)
-                            };
-
                         let field_count = if RTCaches::caches_enabled() {
                             let cached_field_count =
                                 &frame_cache.per_instruction_cache[self.pc as usize];
                             if let PerInstructionCache::Pack(ref field_count) = cached_field_count {
                                 *field_count
                             } else {
-                                let field_count = get_field_count_charge_gas_and_check_depth()?;
+                                let field_count = self.field_count(*sd_idx);
                                 frame_cache.per_instruction_cache[self.pc as usize] =
                                     PerInstructionCache::Pack(field_count);
                                 field_count
                             }
                         } else {
-                            get_field_count_charge_gas_and_check_depth()?
+                            self.field_count(*sd_idx)
                         };
 
+                        let struct_type = self.get_struct_ty(*sd_idx);
+                        interpreter.check_pack_n_values(
```

Comment:
We only need to do the dynamic check for PackClosure, so this code should not need to be changed. Similar for VecPack.